test/serialization/config/FactoriesTest.js

```diff
@@ -57,27 +55,17 @@ describe('FactoriesTest', function () {
         };
     }
 
-    it('should be configured programmatically', function () {
-        return Client.newHazelcastClient(createConfig(cluster.id))
-            .then(function (cl) {
-                client = cl;
-                let map;
-                return client.getMap('furkan').then(function (mp) {
-                    map = mp;
-                    return map.put('foo', new Foo("elma"));
-                }).then(function () {
-                    return map.put('address', new Address('Sahibiata', 42000, 'Konya', 'Turkey'))
-                }).then(function () {
-                    return map.get('foo');
-                }).then(function (res) {
-                    expect(res.foo).to.be.equal('elma');
-                    return map.get('address');
-                }).then(function (res) {
-                    expect(res.street).to.be.equal('Sahibiata');
-                    expect(res.zipCode).to.be.equal(42000);
-                    expect(res.city).to.be.equal('Konya');
-                    expect(res.state).to.be.equal('Turkey');
-                });
-            });
+    it('should be configured programmatically', async function () {
+        client = await Client.newHazelcastClient(createConfig(cluster.id));
+        const map = await client.getMap('furkan');
+        await map.put('foo', new Foo('elma'));
+        await map.put('address', new Address('Sahibiata', 42000, 'Konya', 'Turkey'));
+        const res = await map.get('foo');
+        expect(res.foo).to.be.equal('elma');
+        const res_1 = await map.get('address');
```

Comment:
I guess we can define `res` as let and reuse it

Reply:
Addressing this in the next commit.